### ISSUE TITLE
prompt_embeds variable is overwritten

### ISSUE THREAD

**Issue by dverdu-freepik** [huggingface/diffusers#4506]
### Describe the bug

In example dreambooth/train_dreambooth_lora_sdxl.py, variable prompt_embeds is overwritten, making its shape increases in each step.

`                
                elems_to_repeat = bsz // 2 if args.with_prior_preservation else bsz

                if not args.train_text_encoder:
                    unet_added_conditions = {
                        "time_ids": add_time_ids.repeat(elems_to_repeat, 1),
                        "text_embeds": unet_add_text_embeds.repeat(elems_to_repeat, 1),
                    }
                    prompt_embeds = prompt_embeds.repeat(elems_to_repeat, 1, 1)
                    model_pred = unet(
                        noisy_model_input,
                        timesteps,
                        prompt_embeds,
                        added_cond_kwargs=unet_added_conditions,
                    ).sample
`

### Reproduction

export MODEL_NAME="stabilityai/stable-diffusion-xl-base-1.0"
export INSTANCE_DIR="dog"
export OUTPUT_DIR="lora-trained-xl"

accelerate launch train_dreambooth_lora_sdxl.py \
  --pretrained_model_name_or_path=$MODEL_NAME  \
  --instance_data_dir=$INSTANCE_DIR \
  --output_dir=$OUTPUT_DIR \
  --mixed_precision="fp16" \
  --instance_prompt="a photo of sks dog" \
  --resolution=1024 \
  --train_batch_size=1 \
  --gradient_accumulation_steps=4 \
  --learning_rate=1e-4 \
  --report_to="wandb" \
  --lr_scheduler="constant" \
  --lr_warmup_steps=0 \
  --max_train_steps=500 \
  --validation_prompt="A photo of sks dog in a bucket" \
  --validation_epochs=25 \
  --seed="0" \

### System Info

- `diffusers` version: 0.20.0.dev0
- Platform: Linux-5.15.0-76-generic-x86_64-with-glibc2.35
- Python version: 3.10.12
- PyTorch version (GPU?): 2.0.1+cu117 (True)
- Huggingface_hub version: 0.16.4
- Transformers version: 4.31.0
- Accelerate version: 0.21.0
- xFormers version: 0.0.20

**Comment by sayakpaul** [comment]
It's now fixed: https://github.com/huggingface/diffusers/pull/4509
